lib/Sema/SemanticResolver.cpp

```diff
@@ -2166,6 +2166,14 @@ void SemanticResolver::validateAndDeclareIdentifier(
             "Can't create duplicate variable that shadows a global property: '") +
             ident->_name->str() + "'");
   }
+    
+  if (semCtx_.getDeclarationDecl(ident) &&
+      functionContext()->promotedFuncDecls.count(ident->_name)) {
+    decl = semCtx_.newDeclInScope(ident->_name, kind, curScope_);
+    bindingTable_.put(ident->_name, Binding{decl, ident});
```

Comment:
This needs a comment explaining what the special case is doing.

e.g. Why is it using `put` to override instead of `try_emplace` as below?